sycl/include/CL/sycl/INTEL/dot_product.hpp

```diff
@@ -0,0 +1,76 @@
+//==----------- dot_product.hpp ------- SYCL dot-product -------------------==//
```

Comment:
could you please add documentation for the feature together with the feature?

Reply:
Added documentation: SYCL_ONEAPI_dot_accumulate.asciidoc in sycl/doc/extensions/DotProduct directory.